lib/puppet/parser/environment_compiler.rb

```diff
@@ -0,0 +1,85 @@
+require 'puppet/parser/compiler'
+
+class Puppet::Parser::EnvironmentCompiler < Puppet::Parser::Compiler
+  def self.compile(env)
+    $env_module_directories = nil
+    env.check_for_reparse
+
+    node = Puppet::Node.new(env)
+    node.environment = env
+    new(node).compile
+  rescue => detail
+    message = "#{detail} in environment #{env.name}"
+    Puppet.log_exception(detail, message)
+    raise Puppet::Error, message, detail.backtrace
+  end
+
+  def compile
+    Puppet.override(@context_overrides, "For compiling environment #{environment.name}") do
```

Comment:
Nm .. just realized that you just wanted the additional `catalog` there; changed